streaming guide refactor

Adachi91 opened the pull request:
Made it so that it's a bit more digestible to read and find certain content.

* Fixed a mobile overflow on the X axis issue due to the recommended settings table, this is now confined to accordion's body, with style `overflow-x: auto;` to keep the viewport width from being expanded on mobile.
* Added template for `accordion` (the individual segments) and `accordion group` (Container), instead of creating a lot of html files so everything can be kept in body with markdown still in use.
* Added template for lazy loading images.
* Added webp version of images to srcset to load first if the browser supports them, and falls back to the jpg/png if not.
* Made external links (External being non-JoystickTV) noref, nofollow, and target _blank on all of them, so the user doesn't accidentally navigate away then have to come back to where they were reading.
* Fixed typo's, and rephrased a few things to be more concise and flow more smoothly.
* [JavaScript] Added a small function in `script.js` for assisting in page smooth anchor jumping (e.g. Lovense Integration has a jump to a section under Multistreaming->Stream Master).
* [scss] Added style for the accordions and accordion group, with great pain, I found a solution on SO for the transition to keep the `+`, `-` pixel perfect.

Global:
* [scss:272] Fixed Anchor offset issue for both sidebar ToC and any body-content hashes (#Jump-To)
* [JavaScript:125] Fixed a bug with the `setColorTheme` on page load function, the logic was a bit off so it never remembered your localStorage preference.
* Added lazy loading to youtube embed on `streamer_support.md` (it does not stop all requests on page load, but does lower them until the video is in viewport range)
* Default.html - Removed a rogue `<div>`, which did not seem to serve any purpose, it had no pseudo elements on it, nor classes, even on render. Nothing appears to have changed by removing it.


Cheers.

jwoertink commented:
Thanks for putting this together! It's a big PR so I'll need to take some time to go through it, but I'll try to get it merged in next week, or at least fully reviewed.

Adachi91 commented:
No rush, I expect there might be some changes made too, I went over it a few times, but I may have missed something that fresh eyes might spot, or other changes. 

Have a good weekend !

jwoertink commented on _config.yml:
```diff
@@ -27,6 +27,8 @@ sidebar:
       text: Community Guidelines
     - url: https://joystick.tv/trademark-policy
       text: Trademark Policy
+    - url: https://joystick.tv/branding
+      text: Branding Toolkit
```
good catch :joy:

jwoertink commented on live_streaming.md:
```diff
@@ -68,413 +68,667 @@ software on your device. The streaming software will allow you
+
+{% include lazy_image.html src="/assets/png/obs-setup-stream-service.png" srcset="/assets/webp/obs-setup-stream-service.webp" caption="OBS Service Setup" loading="lazy" %}
+
+> **Note:** If you see "RTMP" for Server, you're Joystick's **older North American server**. To use another server, select **"Custom"** from the Service list and enter the **Server URL** from your Joystick Stream Settings page.
```
I think it's been long enough that we can get rid of this line.

jwoertink commented on live_streaming.md:
```diff
@@ -68,413 +68,667 @@ software on your device. The streaming software will allow you
+Example:
+- Streaming at **1080p@60** to **Joystick + 2 other platforms** requires **a minimum of 30 Mbps upload** (~8 Mbps per site).
 
 ## Mobile Streaming
 
```
Not necessary for this PR, but just a consideration: Should we also add the "in-browser" streaming option with information on using that along with limitations of using browser VS a mobile app?

jwoertink commented on live_streaming.md:
```diff
@@ -68,413 +68,667 @@ software on your device. The streaming software will allow you
 
-After installing, you can open the settings in the the Docks menu of OBS under the **"Multiple Output"** option.
+### **Reputable Cloud Multistream Services**
+🚧 **Note:** Most mainstream multistreaming services prohibit adult content, as payment processors and financial institutions classify it as "high risk." If you know of any services that allow it, feel free to suggest them [here](https://github.com/joysticktv/joysticktv.github.io/issues){:target="_blank"}.
```
The github link is nice, but I'd imagine most people reading the guides won't have an account. Maybe we say post the suggestion on our Discord?

jwoertink commented on live_streaming.md:
```diff
@@ -68,413 +68,667 @@ software on your device. The streaming software will allow you
+{% capture Aitum_Multistream_body %}
+[Aitum Multistream](https://github.com/Aitum/obs-aitum-multistream){:rel="noopener nofollow noreferrer" target="_blank"} is a **simpler alternative** to multi-RTMP in OBS, offering an **easier UI** for managing multiple stream outputs.
+
+### **How to Set Up Aitum Multistream**
```
Nice! Never heard of this one before

jwoertink reviewed (commented):
This is really awesome. Great work on this. I love how you componentized some of the bits too. I left a few comments, but I think this is just about ready to go.

Adachi91 commented:
Updated with the PR recommendations and included guide for mobile and desktop in-browser streaming.

I also changed JoystickTV references to the legal naming `Joystick` (Service).

jwoertink commented:
Thanks for the updates!
